File: source/android/Fortumo.java
/*
java implementation of the Fortumo extension.

Add android-specific functionality here.

These functions are called via JNI from native code.
*/
/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */
import android.content.Intent;
import com.ideaworks3d.marmalade.LoaderAPI;
import com.ideaworks3d.marmalade.LoaderActivity;
import android.content.Context;

import com.fortumo.android.*;
import com.fernsblossom.*;

class Fortumo
{
    public int FortumoRequestPayment(boolean isConsumable, int creditsMultiplier, String displayString, String productName, String serviceId, String serviceSecret, String SKU)
    {
		//System.out.println("FortumoRequestPayment:\nisConsumable="+isConsumable+"\ncreditsMultiplier="+creditsMultiplier+"\ndisplayString="+displayString+"\nproductName="+productName+"\nserviceId="+serviceId+"\nserviceSecret="+serviceSecret+"\nSKU="+SKU);
		PaymentRequestBuilder builder = new PaymentRequestBuilder();
		builder.setConsumable(isConsumable);
		builder.setCreditsMultiplier(creditsMultiplier/100.0f);
		if (displayString!=null)
			builder.setDisplayString(displayString);
		if (productName!=null)
			builder.setProductName(productName);
		if (serviceId!=null && serviceSecret!=null)
			builder.setService(serviceId,serviceSecret);
		if (SKU!=null)
			builder.setSku(SKU);
		com.fernsblossom.PayActivity.handle=null;
		Intent intent = new Intent(LoaderActivity.m_Activity, com.fernsblossom.PayActivity.class);
		LoaderActivity.m_Activity.startActivity(intent);
		com.fernsblossom.PayActivity.pay(builder);
        return 0;
    }
    public void FortumoTerminate()
    {
		//com.fernsblossom.PayActivity.handle.finish();
		//System.out.println("Fortumo terminated");
    }
    public int FortumoInit()
    {
		//if (com.fernsblossom.PayActivity.handle==null) {
		//}
		//System.out.println("Fortumo initiated");
        return 0;
    }
}

File: source/generic/Fortumo.cpp
/*
Generic implementation of the Fortumo extension.
This file should perform any platform-indepedentent functionality
(e.g. error checking) before calling platform-dependent implementations.
*/

/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */


#include "Fortumo_internal.h"
s3eResult FortumoInit()
{
    //Add any generic initialisation code here
    return FortumoInit_platform();
}

void FortumoTerminate()
{
    //Add any generic termination code here
    FortumoTerminate_platform();
}

s3eResult FortumoRequestPayment(bool isConsumable, int creditsMultiplier, const char* displayString, const char* productName, const char* serviceId, const char* serviceSecret, const char* SKU)
{
	return FortumoRequestPayment_platform(isConsumable, creditsMultiplier, displayString, productName, serviceId, serviceSecret, SKU);
}
